lib/torrent.js

```diff
@@ -618,12 +624,18 @@ class Torrent extends EventEmitter {
             this._debug('piece verified %s', index)
             this._markVerified(index)
           } else {
+            this.pieces[index] = new Piece(piece.length || this.pieceLength)
+            this.bitfield.set(index, false)
+            this.select(index, index, 1)
+            this.files.forEach(file => {
+              if (file.done && file.includes(index)) file.done = false
+            })
```

Comment:
Ok, I agree.

Then, I think we should create a separate private method (called something like `markUnverified`) with the lines for `pieces` and `bitfield`.

And then we should change the code from `_checkDone` to mark the torrent as not done and mark the files as not done.

I am not sure if this is breaking change or a fix, since the torrent will change state without the user knowing. @feross @jhiesey ?

Reply:
ok I'll put it back... I just remember I added it for a reason, but the reason escapes me now.    I'll retest with it put back and removed from this method

Reply:
ok I'll put it back... I just remember I added it for a reason, but the reason escapes me now. I'll retest with it put back and removed from this method

Reply:
do you know if select is needed? i wasn't sure

Reply:
I don't think I'm a fan of marking the files as not done in _checkdone.
I had a pr (not sure if it's merged) where I wanted to see events as the pieces are verified or not...
The files and pieces will be out of sync if we wait till we are done with the file.  And I'm not sure what will happen when the client sees that it's missing pieces, but the files are all marked done.

Each verification is an async operation.... there's room for other code to run in between.

Reply:
@alxhotel can we resolve this?